### PR TITLE
CORE-230: Add limit to findAll function

### PR DESCRIPTION
**Pull request by cholcombe973**
This adds an optional limit parameter to the `findAll` function allowing callers to limit the results. Getting the last message now will be a `findall(query: "..", limit: 1)` function call.

### CODE REVIEW

**Review by tsmbl — CHANGES_REQUESTED** [review]
Hey, looks good, a couple of comments

1. Should we also update endpoint query?
2. If there's an endpoint query, let's prob move validation to dto via @Min decorator?
3. Let's append jira ticket name also in PR name in the next PRs, wdyt?

File: src/dialect/dialect.controller.dto.ts [hunk 1]
```diff
@@ -96,6 +98,12 @@ export class CreateDialectMemberDto {
   readonly scopes!: MemberScopeDto[];
 }
 
+export class PositiveNumberDto {
+  @IsNumberString()
+  @Min(1)
+  readonly value!: string;
```
Comment (tsmbl) [inline src/dialect/dialect.controller.dto.ts:104]: Hey, should it be number and optional for backward compatibility sake?

e.g. example of how paging is done in FindDappMessagesQueryDto 
image

File: src/dialect/dialect.service.ts [hunk 2]
```diff
@@ -92,7 +92,11 @@ export class DialectService {
 
   async findAll(
     query: FindDialectQuery,
+    limit: number = 100,
   ): Promise<MemberedAndMessagedDialect[]> {
+    if (limit < 1) {
```
Comment (tsmbl) [inline src/dialect/dialect.service.ts:97]: do we need this explicit validation if `@Min()` decorator is applied?

File: src/dialect/dialect.controller.ts [hunk 3]
```diff
@@ -126,6 +127,7 @@ export class DialectController {
   async findAll(
     @AuthPrincipal() { wallet }: Principal,
     @Query() query: FindDialectQueryDto,
+    @Param('limit') limit: PositiveNumberDto,
```
Comment (tsmbl) [inline src/dialect/dialect.controller.ts:130]: I think this parameter should be included in query  `FindDialectQueryDto`, wdyt?

**Review by tsmbl — CHANGES_REQUESTED** [review]
Nice, added several comments/questions

File: src/dialect/dialect.controller.dto.ts (hunk 1 above)
Comment (cholcombe973) [reply src/dialect/dialect.controller.dto.ts:104]: Neat! I like that.

File: src/dialect/dialect.service.ts (hunk 2 above)
Comment (cholcombe973) [reply src/dialect/dialect.service.ts:97]: That's true. The validator should take care of it. I'll remove it.

File: src/dialect/dialect.controller.ts (hunk 3 above)
Comment (cholcombe973) [reply src/dialect/dialect.controller.ts:130]: That sounds interesting. How would that work?

File: src/dialect/dialect.controller.dto.ts [hunk 4]
```diff
@@ -116,6 +120,12 @@ export class FindDialectQueryDto {
   @ArrayNotEmpty()
   @IsPublicKey({ each: true })
   readonly memberPublicKeys?: string[];
+  @IsInt()
+  @IsOptional()
+  @Min(1)
+  @Max(500)
+  @Type(() => Number)
+  readonly take?: number;
```
Comment (tsmbl) [inline src/dialect/dialect.controller.dto.ts:128]: I think it's ambiguous how this parameter is applied: to list of threads or to messages? Probably we can 1) extract pagination dto to be later used in other system modules 2) change dto to be explicit on what is paged. What do you think?

Comment (tsmbl) [inline src/dialect/dialect.controller.dto.ts:128]: I think we should also use take parameter in other endpoints where the query is used, wdyt?

**Review by tsmbl — CHANGES_REQUESTED** [review]
Awesome, left a couple of comments / questions

File: src/dialect/dialect.controller.dto.ts [hunk 5]
```diff
@@ -6,8 +6,11 @@ import {
...
   IsArray,
   IsBoolean,
+  IsInt,
   IsOptional,
   IsString,
+  Max,
+  Min,
   ValidateNested,
 } from 'class-validator';
 import { Type } from 'class-transformer';
```
[the remark this answers is not in the document]
Comment (cholcombe973) [reply src/dialect/dialect.controller.dto.ts:16]: Yeah Max/Min is used down at line 125/126

File: src/dialect/dialect.controller.dto.ts (hunk 4 above)
Comment (cholcombe973) [reply src/dialect/dialect.controller.dto.ts]: The consensus over slack DM's was that this is a bit of scope creep so we're going to push this change off to a future PR.

**Review by tsmbl — APPROVED** [review]
Awesome, lgtm!

**Pull request merged**